autoload/sj/elixir.vim

```diff
@@ -189,3 +189,85 @@ function! sj#elixir#JoinArray()
 
   return 1
 endfunction
+
+function! sj#elixir#SplitPipe()
+  let line = getline('.')
+
+  if line =~ '^\s*|>\s\+'
+    return 0
+  endif
+
+  call sj#PushCursor()
+  normal! 0f(
+  let [function_name, function_start, function_end, function_type] =
+        \ sj#argparser#elixir#LocateFunction()
+  call sj#PopCursor()
+
+  if function_start < 0
+    return 0
+  endif
+
+  let parser = sj#argparser#elixir#Construct(function_start, function_end, line)
+  call parser.Process()
+
+  let args = parser.args
+
+  let function_call = sj#Trim(strpart(line, 0, function_start - 2))
+  let result = args[0] . "\n|> " . function_call . '(' . join(args[1:], ', ') . ')'
```

Comment:
Trying this out with `String.length()` rather than `foo()`, I get:

``` elixir
"one"
|> String.length() + String.length("two")
```

Output:

```
** (ArgumentError) cannot pipe "one" into String.length() + String.length("two"), the :+ operator can only take one argument
    (elixir 1.11.2) lib/macro.ex:290: Macro.pipe/3
    (stdlib 3.14) lists.erl:1267: :lists.foldl/3
    (elixir 1.11.2) expanding macro: Kernel.|>/2
    test.exs:2: (file)
```

The following works (and that's how `mix format` formats it):

``` elixir
("one"
 |> String.length()) + String.length("two")
```

But I've no idea why anyone would want to do that!

I think we're better off trying to detect this scenario and refusing to do anything? It doesn't really seem like an important use case to support.

Reply:
Yeah, agreed :+1: